### PR TITLE
feat: extend transactions to cover after and beforeOperation hooks

### PR DESCRIPTION
**Pull request by DanRibbens**
## Description

This change includes the following improvements:
- All operations that call beforeOperation begins a transaction before the hooks are called
- Adds `req` to the beforeOperation hook args
- Adds `req` to the afterOperation hook 

- [x] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Existing test suite passes locally with my changes
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: docs/hooks/collections.mdx [hunk 1]
```diff
@@ -75,6 +75,7 @@ import { CollectionBeforeOperationHook } from 'payload/types'
 const beforeOperationHook: CollectionBeforeOperationHook = async ({
   args, // original arguments passed into the operation
   operation, // name of the operation
+  req, // full express request
```
Comment (denolfe) [inline docs/hooks/collections.mdx:78]: This should specify that it is a `PayloadRequest`, not expess. Also, can we make these proper JSDoc comments that will show in the IDE?

File: docs/hooks/collections.mdx [hunk 2]
```diff
@@ -209,6 +210,7 @@ import { CollectionAfterOperationHook } from 'payload/types'
 const afterOperationHook: CollectionAfterOperationHook = async ({
   args, // arguments passed into the operation
   operation, // name of the operation
+  req, // full express request
```
Comment (denolfe) [inline docs/hooks/collections.mdx:213]: Same `PayloadRequest` and JSDoc as above.

File: docs/hooks/collections.mdx (hunk 1 above)
Comment (DanRibbens) [reply docs/hooks/collections.mdx:78]: Nope, this is just documentation. We're in an `.mdx` here.

File: docs/hooks/collections.mdx (hunk 2 above)
Comment (DanRibbens) [reply docs/hooks/collections.mdx:213]: .mdx only, not actual code :)

**Pull request merged**